app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -238,24 +240,26 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         }
 
         pocketVideoMegaTileView.setOnClickListener(this)
+    }
 
-        if (pocketVideos.isCompleted) {
-            pocketVideoMegaTileView.pocketVideos = pocketVideos.getCompleted()
-
-            if (pocketVideoMegaTileView.pocketVideos == null) {
-                showMegaTileError()
-            } else {
-                pocketVideosContainer.visibility = View.VISIBLE
-                pocketErrorContainer.visibility = View.GONE
-                pocketVideosContainer.contentDescription = resources.getString(R.string.pocket_home_a11y_tile_focused,
-                        resources.getString(R.string.pocket_brand_name))
-            }
-        } else {
-            // TODO: #864 show loading screen
-            launch(UI) {
-                pocketVideoMegaTileView.pocketVideos = pocketVideos.await()
+    fun observeForMegaTile(fragment: Fragment) {
+        // TODO remove this
+        // This function is necessary because BrowserNavigationOverlay is not a LifecycleOwner,
+        // and so cannot subscribe to PocketRepo's LiveData. When the overlay is turned into a
+        // fragment, this can be moved into the overlay fragment
+        val factory = fragment.context!!.serviceLocator.viewModelFactory
+        // TODO initialize pocketViewModel in onCreate when overlay is refactored into a fragment
+        pocketViewModel = ViewModelProviders.of(fragment, factory).get(PocketViewModel::class.java)
+
+        pocketViewModel.state.observe(fragment.viewLifecycleOwner, Observer { state ->
+            when (state) {
+                is PocketViewModelState.Error -> showMegaTileError()
+                is PocketViewModelState.Feed -> {
+                    pocketVideoMegaTileView.setContent(state.feed)
+                }
+                null -> return@Observer
```

Comment:
Why would the state be null?

Reply:
Not sure, but `Observer#onChanged` passes a nullable argument.  Possibly if an object observes before one has been set, I'm not sure how LiveData handles that case.